src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFChart.java

```diff
@@ -1,123 +1,343 @@
-import org.openxmlformats.schemas.drawingml.x2006.chart.CTChartSpace;
-
-/**
- * Represents a Chart in a .docx file
- */
-@Beta
-public class XWPFChart extends XDDFChart {
-
-    // lazy initialization
-    private Long checksum;
-
-    /**
-     * Construct a chart from a package part.
-     *
-     * @param part the package part holding the chart data,
-     * the content type must be <code>application/vnd.openxmlformats-officedocument.drawingml.chart+xml</code>
-     *
-     * @since POI 4.0.0
-     */
-    protected XWPFChart(PackagePart part) throws IOException, XmlException {
-        super(part);
-    }
-
-    @Override
-    protected void commit() throws IOException {
-        XmlOptions xmlOptions = new XmlOptions(DEFAULT_XML_OPTIONS);
-        xmlOptions.setSaveSyntheticDocumentElement(new QName(CTChartSpace.type.getName().getNamespaceURI(), "chartSpace", "c"));
-
-        try (OutputStream out = getPackagePart().getOutputStream()) {
-            chartSpace.save(out, xmlOptions);
-        }
-    }
-
-    public Long getChecksum() {
-        if (this.checksum == null) {
-            InputStream is = null;
-            byte[] data;
-            try {
-                is = getPackagePart().getInputStream();
-                data = IOUtils.toByteArray(is);
-            } catch (IOException e) {
-                throw new POIXMLException(e);
-            } finally {
-                try {
-                    if (is != null) {
-                        is.close();
-                    }
-                } catch (IOException e) {
-                    throw new POIXMLException(e);
-                }
-            }
-            this.checksum = IOUtils.calculateChecksum(data);
-        }
-        return this.checksum;
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-        /**
-         * In case two objects ARE equal, but its not the same instance, this
-         * implementation will always run through the whole
-         * byte-array-comparison before returning true. If this will turn into a
-         * performance issue, two possible approaches are available:<br>
-         * a) Use the checksum only and take the risk that two images might have
-         * the same CRC32 sum, although they are not the same.<br>
-         * b) Use a second (or third) checksum algorithm to minimise the chance
-         * that two images have the same checksums but are not equal (e.g.
-         * CRC32, MD5 and SHA-1 checksums, additionally compare the
-         * data-byte-array lengths).
-         */
-        if (obj == this) {
-            return true;
-        }
-
-        if (obj == null) {
-            return false;
-        }
-
-        if (!(obj instanceof XWPFChart)) {
-            return false;
-        }
-        return false;
-    }
-
-    @Override
-    public int hashCode() {
-        return getChecksum().hashCode();
-    }
-}
+/* ====================================================================

```

Comment:
I suspect there is a line ending issue with this file. 

Could you check that this version of the file has Unix-like CR line ending instead of Windows CRLF?

Reply:
https://help.github.com/articles/dealing-with-line-endings/ might help